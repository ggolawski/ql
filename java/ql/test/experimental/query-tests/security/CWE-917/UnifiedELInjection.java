import javax.el.ELProcessor;
import javax.el.ExpressionFactory;
import javax.el.MethodExpression;
import javax.el.ValueExpression;

import org.springframework.web.bind.annotation.RequestParam;

public class UnifiedELInjection {
  public void testMethodExpression(@RequestParam String expr) {
    MethodExpression me = ExpressionFactory.newInstance().createMethodExpression(null, expr, null, null);
    me.invoke(null, null);
  }

  public void testValueExpression(@RequestParam String expr) {
    ExpressionFactory ef = ExpressionFactory.newInstance();
    ValueExpression ve = ef.createValueExpression(null, expr, null);
    ve.getValue(null);
    ve.setValue(null, null);
  }

  public void testELProcessor(@RequestParam String expr) {
    ELProcessor proc = new ELProcessor();
    proc.eval(expr);
    proc.getValue(expr, null);
    proc.setValue(expr, null);
  }
}
