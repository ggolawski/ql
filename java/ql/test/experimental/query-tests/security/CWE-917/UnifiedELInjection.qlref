Security/CWE/CWE-917/UnifiedELInjection.ql
