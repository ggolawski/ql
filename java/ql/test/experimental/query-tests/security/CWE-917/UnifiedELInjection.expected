edges
| UnifiedELInjection.java:9:36:9:60 | expr : String | UnifiedELInjection.java:11:5:11:6 | me |
| UnifiedELInjection.java:14:35:14:59 | expr : String | UnifiedELInjection.java:17:5:17:6 | ve |
| UnifiedELInjection.java:14:35:14:59 | expr : String | UnifiedELInjection.java:18:5:18:6 | ve |
| UnifiedELInjection.java:21:31:21:55 | expr : String | UnifiedELInjection.java:23:15:23:18 | expr |
| UnifiedELInjection.java:21:31:21:55 | expr : String | UnifiedELInjection.java:24:19:24:22 | expr |
| UnifiedELInjection.java:21:31:21:55 | expr : String | UnifiedELInjection.java:25:19:25:22 | expr |
nodes
| UnifiedELInjection.java:9:36:9:60 | expr : String | semmle.label | expr : String |
| UnifiedELInjection.java:11:5:11:6 | me | semmle.label | me |
| UnifiedELInjection.java:14:35:14:59 | expr : String | semmle.label | expr : String |
| UnifiedELInjection.java:17:5:17:6 | ve | semmle.label | ve |
| UnifiedELInjection.java:18:5:18:6 | ve | semmle.label | ve |
| UnifiedELInjection.java:21:31:21:55 | expr : String | semmle.label | expr : String |
| UnifiedELInjection.java:23:15:23:18 | expr | semmle.label | expr |
| UnifiedELInjection.java:24:19:24:22 | expr | semmle.label | expr |
| UnifiedELInjection.java:25:19:25:22 | expr | semmle.label | expr |
#select
| UnifiedELInjection.java:11:5:11:6 | me | UnifiedELInjection.java:9:36:9:60 | expr : String | UnifiedELInjection.java:11:5:11:6 | me | Unified EL expression might include input from $@. | UnifiedELInjection.java:9:36:9:60 | expr | this user input |
| UnifiedELInjection.java:17:5:17:6 | ve | UnifiedELInjection.java:14:35:14:59 | expr : String | UnifiedELInjection.java:17:5:17:6 | ve | Unified EL expression might include input from $@. | UnifiedELInjection.java:14:35:14:59 | expr | this user input |
| UnifiedELInjection.java:18:5:18:6 | ve | UnifiedELInjection.java:14:35:14:59 | expr : String | UnifiedELInjection.java:18:5:18:6 | ve | Unified EL expression might include input from $@. | UnifiedELInjection.java:14:35:14:59 | expr | this user input |
| UnifiedELInjection.java:23:15:23:18 | expr | UnifiedELInjection.java:21:31:21:55 | expr : String | UnifiedELInjection.java:23:15:23:18 | expr | Unified EL expression might include input from $@. | UnifiedELInjection.java:21:31:21:55 | expr | this user input |
| UnifiedELInjection.java:24:19:24:22 | expr | UnifiedELInjection.java:21:31:21:55 | expr : String | UnifiedELInjection.java:24:19:24:22 | expr | Unified EL expression might include input from $@. | UnifiedELInjection.java:21:31:21:55 | expr | this user input |
| UnifiedELInjection.java:25:19:25:22 | expr | UnifiedELInjection.java:21:31:21:55 | expr : String | UnifiedELInjection.java:25:19:25:22 | expr | Unified EL expression might include input from $@. | UnifiedELInjection.java:21:31:21:55 | expr | this user input |
